app/src/lib/menu-update.ts

```diff
@@ -164,12 +166,29 @@ function getMenuState(state: IAppState): Map<MenuIDs, IMenuItemState> {
     menuStateBuilder.disable('pull')
   }
 
+  const welcomeScopedIds: ReadonlyArray<MenuIDs> = [
+    'new-repository',
+    'add-local-repository',
+    'clone-repository',
+    'create-branch',
+  ]
+
+  if (inWelcomeFlow) {
+    for (const id of welcomeScopedIds) {
+      menuStateBuilder.disable(id)
+    }
+  } else {
+    for (const id of welcomeScopedIds) {
+      menuStateBuilder.enable(id)
```

Comment:
So it seems to me like this would unconditionally enable `create-branch` even though the `TipState` is unknown which is probably not what we want.

Additionally (and this is an edge case, I'll grant you). It's possible to both have a repository selected and be in the welcome flow at the same time by clearing the `localStorage` key so I think we should make `repositoryActive` take `inWelcomeFlow` into account

```ts
const repositoryActive = windowOpen && repositorySelected && !state.showWelcomeFlow
```

By doing that and removing the `create-branch` I think we'd get rid of this particular problem. This logic is pretty hard to reason about. Anytime we have an intersection of items in both the `repositoryScopedIds` and `inWelcomeFlow` we're in for a bad time. I don't have a better solution for now though but perhaps a warning comment would be in order?